packages/techdocs-common/__mocks__/@azure/storage-blob.ts

```diff
@@ -48,20 +50,45 @@ export class BlockBlobClient {
   }
 
   uploadFile(source: string): Promise<BlobUploadCommonResponse> {
-    return Promise.resolve({
-      _response: {
-        request: {
-          url: `https://example.blob.core.windows.net`,
-        } as any,
-        status: 200,
-        headers: {} as any,
-      },
+    return new Promise((resolve, reject) => {
+      if (!fs.existsSync(source)) {
+        reject(`The file ${source} does not exist`);
+      } else {
+        resolve({
+          _response: {
+            request: {
+              url: `https://example.blob.core.windows.net`,
+            } as any,
+            status: 200,
+            headers: {} as any,
+          },
+        });
+      }
```

Comment:
You don't need to create a new Promise. It's much simpler to just say,
```ts
if (!fs.existsSync(source)) {
  return Promise.reject(`The file ${source} does not exist`);
} else {
  return Promise.resolve({
    _response: {
      request: {
        url: `https://example.blob.core.windows.net`,
      } as any,
      status: 200,
      headers: {} as any,
    },
  });
}
```

Reply:
Alternatively, you could make it an async function.

```ts
  async uploadFile(source: string): Promise<BlobUploadCommonResponse> {
     if (!fs.existsSync(source)) {
        throw `The file ${source} does not exist`;
      } else {
        return {
          _response: {
            request: {
              url: `https://example.blob.core.windows.net`,
            } as any,
            status: 200,
            headers: {} as any,
          },
        }
      }
  }
```

Also, you should probably throw an error `new Error(\`The file ${source} does not exist\`)`

Reply:
Do you think that would solve it?
``` js
  uploadFile(source: string): Promise<BlobUploadCommonResponse> {
    if (!fs.existsSync(source)) {
      return Promise.reject(new Error(`The file ${source} does not exist`));
    } else {
      return Promise.resolve({
        _response: {
          request: {
            url: `https://example.blob.core.windows.net`,
          } as any,
          status: 200,
          headers: {} as any,
        },
      });
    }
  }
```

Reply:
Yeah, that should do it